catalog/core/catalog-core-eventcommands/src/main/java/org/codice/ddf/catalog/pubsub/command/ListCommand.java

```diff
@@ -73,46 +80,95 @@
 
   @Override
   public Object execute() throws Exception {
-
     PrintStream console = System.out;
 
-    Map<String, ServiceReference<Subscription>> subscriptionIds = getSubscriptions(id, ldapFilter);
-    if (subscriptionIds.size() == 0) {
+    Map<String, ServiceReference<Subscription>> subscriptions = getSubscriptions(id, ldapFilter);
+    List<SubscriptionListEntry> entries =
+        subscriptions
+            .entrySet()
+            .stream()
+            .map(SubscriptionListEntry::new)
+            .collect(Collectors.toList());
+
+    if (entries.isEmpty()) {
       console.println(RED_CONSOLE_COLOR + NO_SUBSCRIPTIONS_FOUND_MSG + DEFAULT_CONSOLE_COLOR);
     } else {
-      console.println();
-      console.print("Total subscriptions found: " + subscriptionIds.size());
-      console.println();
-      console.println();
-      console.print(CYAN_CONSOLE_COLOR);
-      console.print(SUBSCRIPTION_ID_COLUMN_HEADER);
-      console.print("\t\t| ");
-      console.print(CALLBACK_COLUMN_HEADER);
-      console.println(DEFAULT_CONSOLE_COLOR);
-      subscriptionIds.entrySet().forEach(this::printSubscription);
+      printSubscriptions(entries);
     }
 
     return null;
   }
 
-  private void printSubscription(Map.Entry<String, ServiceReference<Subscription>> entry) {
-
+  private void printSubscriptions(List<SubscriptionListEntry> entries) {
     PrintStream console = System.out;
-    Subscription subscription = bundleContext.getService(entry.getValue());
-    String rowColor = "";
 
-    if (subscription != null
-        && subscription.getDeliveryMethod() instanceof Pingable
-        && !((Pingable) subscription.getDeliveryMethod()).ping()) {
-      rowColor = RED_CONSOLE_COLOR;
-    }
+    int[] columnWidth = new int[4];
+    columnWidth[0] = entries.stream().mapToInt(s -> s.id.length()).max().orElse(-1);
```

Comment:
✏️ A loop would be more efficient here than these four separate reduce operations.